biojava-structure/src/main/java/org/biojava/nbio/structure/Calc.java

```diff
@@ -601,12 +601,14 @@ public static final void translate(Atom atom, Vector3d v) {
 	 * @param v
 	 */
 	public static final void translate(Group group, Vector3d v) {
-		AtomIterator iter = new AtomIterator(group);
 
-		while (iter.hasNext()) {
-			Atom atom = iter.next();
+		for (Atom atom : group.getAtoms()) {
 			translate(atom, v);
-
+		}
+		for (Group altG : group.getAltLocs()) {
+			for (Atom atom : altG.getAtoms()) {
```

Comment:
Why not using `translate(g, v)`? It would be a recursive call to this function, nicer solution.